src/cli/cli.cpp

```diff
@@ -3202,7 +3202,8 @@ void Interpreter::HandlePingStatistics(const otPingSenderStatistics *aStatistics
 {
     OutputFormat("%u packets transmitted, %u packets received.", aStatistics->mSentCount, aStatistics->mReceivedCount);
 
-    if ((aStatistics->mSentCount != 0) && !aStatistics->mIsMulticast)
+    if ((aStatistics->mSentCount != 0) && !aStatistics->mIsMulticast &&
+        aStatistics->mReceivedCount <= aStatistics->mSentCount)
```

Comment:
```suggestion
    if (aStatistics->mSentCount != 0)
```

Should we treat `mReceivedCount` > `mSentCount` as if it's `mSentCount`? (i.e. 0% loss)
and avoid special-casing mutlicast ?

Reply:
I think `mReceivedCount > mSentCount` doesn't imply 0% packet loss. So far I can think of two cases:
- In unicast case, when something's not working properly, the destination sends 2 replies for every request it receives. Then the `mReceivedCount` will be twice as much as `mSentCount`, assuming there's no packet loss. However, even if there are some packet lost, the `mReceivedCount` could still be greater than `mSentCount`.

Reply:
I think we should try to print the `Packet loss` as long as it's possible. 

> In unicast case, when something's not working properly, the destination sends 2 replies for every request it receives.

I think it's an abnormal case that we need to fix. `mReceivedCount` should not be greater than `mSendCount` for unicast ping.

> In multicast case, when there are two nodes receiving echo requests, the mReceivedCount could become greater than mSentCount easily when the packet loss rate is < 50%.

It would be better if we can handle the multicast case correctly by showing duplicates., e.g. like below:
```
$ ping 224.0.0.1
PING 224.0.0.1 (224.0.0.1) 56(84) bytes of data.
64 bytes from 100.84.187.252: icmp_seq=1 ttl=64 time=0.578 ms
64 bytes from 100.84.187.253: icmp_seq=1 ttl=64 time=0.627 ms
64 bytes from 100.84.187.253: icmp_seq=2 ttl=64 time=0.560 ms
64 bytes from 100.84.187.252: icmp_seq=2 ttl=64 time=0.560 ms
^C
--- 224.0.0.1 ping statistics ---
2 packets transmitted, 2 received, +2 duplicates, 0% packet loss, time 1019ms
rtt min/avg/max/mdev = 0.560/0.581/0.627/0.027 ms
```

But I am fine to leave for future enhancement.